uat-web-app/data_generator.py

```diff
@@ -73,60 +79,190 @@ def retrieve_alpha_page_data(uat_id, alpha_terms, html_tree):
         "alpha": alpha_terms,
         "gtype": view_type,
         "element": element,
-        "alphabet": alphabet
+        "alphabet": alphabet,
+        "sort": sort_direction,
     }
 
 
-def search_terms(lookup_term, alpha_terms):
+def normalize_term(term):
+    """
+    Normalizes a term for comparison:
+    - Lowercases
+    - Removes common punctuation
+    - Converts plurals to singular using inflect
+    """
+    term = term.lower()
+    term = re.sub(r"[\s\-_'\"/.,!?;:()“”’‘]", "", term)
```

Comment:
There might be an unescaped `/` in this regex. If you're looking to substitute `/`, I think it needs to be `[\s\-_'\"\/.,!?;:()“”’‘]`